lib/vacuum/locale.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Vacuum
+  # The target Amazon Locale
+  class Locale
+    NotFound = Class.new(ArgumentError)
+
+    attr_reader :code, :host, :region
+
+    def self.find(code)
+      code = code.to_s.upcase
+      code = 'GB' if code == 'UK'
```

Comment:
I would prefer using symbols over strings for locale code. 

Their definition matches perfectly to this use case - e.g. “scalar value objects used as identifiers, mapping immutable strings to fixed internal values.”

That last part _fixed internal values_ has a very subtle difference on how those are stored in memory.

Every time a string is written, it creates a new object with a new place in memory, even if the string is textually identical to an existing string. However, since symbols are immutable, they always refer to the same object and the same place in memory.

Since string allocation is such a huge pain in Ruby, I think it's really important to stick with symbols (e.g. :gb, :uk)